Specs/Scene/ExpressionSpec.js

```diff
@@ -499,6 +501,140 @@ defineSuite([
+        var expression = new Expression('color(\'#ffffff\')["red"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(1);
+
+        expression = new Expression('rgb(255, 255, 0)["green"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(1);
+
+        expression = new Expression('color("cyan")["blue"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(1);
+
+        expression = new Expression('rgba(255, 255, 0, 0.5)["alpha"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(0.5);
+    });
+
+    it('evaluates color properties (["x"], ["y"], ["z"], ["w"])', function() {
+        var expression = new Expression('color(\'#ffffff\')["x"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(1);
+
+        expression = new Expression('rgb(255, 255, 0)["y"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(1);
+
+        expression = new Expression('color("cyan")["z"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(1);
+
+        expression = new Expression('rgba(255, 255, 0, 0.5)["w"]');
+        expect(expression.evaluate(frameState, undefined)).toEqual(0.5);
+    });
+
+    it('evaluates vector', function() {
+        var expression = new Expression('vec2()');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(0.0, 0.0, 0.0, 0.0));
+
+        expression = new Expression('vec3()');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(0.0, 0.0, 0.0, 0.0));
+
+        expression = new Expression('vec4()');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(0.0, 0.0, 0.0, 0.0));
+
+        expression = new Expression('vec2(1.0, 2.0)');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(1.0, 2.0, 0.0, 0.0));
+
+        expression = new Expression('vec3(1.0, 2.0, 3.0)');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(1.0, 2.0, 3.0, 0.0));
+
+        expression = new Expression('vec4(1.0, 2.0, 3.0, 4.0)');
+        expect(expression.evaluate(frameState, undefined)).toEqual(new Cartesian4(1.0, 2.0, 3.0, 4.0));
+    });
+
+    it('evaluates vector with expressions as arguments', function() {
+        var feature = new MockFeature();
+        feature.addProperty('height', 2);
+        feature.addProperty('width', 4);
+        feature.addProperty('depth', 3);
+        feature.addProperty('scale', 1);
+
+        var expression = new Expression('vec4(${height}, ${width}, ${depth}, ${scale})');
+        expect(expression.evaluate(frameState, feature)).toEqual(new Cartesian4(2.0, 4.0, 3.0, 1.0));
+    });
+
+    it('evaluates expression with multiple nested vectors', function() {
+        var expression = new Expression('vec4(vec2(1, 2)[vec3(6, 1, 5).y], 2, vec4().w, 5)');
```

Comment:
This is not allowed, all components must evaluate to numbers (which the example above does). I will make this clear in the spec.